app/1.0.x/plugin-development/custom-entities.md

```diff
@@ -129,76 +129,105 @@ To see a real-life example, give a look at the [Key-Auth plugin migrations](http
+  `id`. If you are using Cassandra and need a composite key, it should have the same
+  fields as the *partition key*.
+* `endpoint_key`: An optional string, which would be the name of a field used as an
+  alternative identifier on the Admin API. On the example above, `key` is the endpoint_key. This
+  means that a credential with `id = 123` and `key = "foo"` could be referenced as
+  `/keyauth_credentials/123` and `/keyauth_credentials/foo`.
+* `cache_key`: An optional array of field names. It is used for generating the `cache_key`, a
+  string which must unequivocally reference the entity. A unique field, like `key` in your example,
+  is usually good candidate, but sometimes it's useful to combine more than 1 field.
+
+And some notes regarding fields:
+
+* The `typedefs` variable (obtained by requiring `kong.db.schema.typedefs`) is a table containing
+  [a lot of useful type definitions and aliases](https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua),
+  including `typedefs.uuid`, the most usual type for the primary key, and `typedefs.auto_timestamp_s`,
+  for `created_at` fields.
+* Schemas support the following scalar types: `string`, `integer`, `number` and `boolean`. These types
+  will need to be backed by fields of appropriately similar types in the database. Compound types like Arrays, Records,
+  or Sets are also supported.
+* Foreign relationships can be specified using the `foreign` type on a field, as exemplified by the
+  `consumer` field in the previous example. The `reference` attribute *must* be the name of an existing
+  schema. This means that if a schema B has a foreign key pointing to schema A, then A needs to be loaded
+  before B.
+* The `on_delete = "cascade"` attribute is specific to foreign relationships, and dictates what happens
+  with the "children" when a "parent" entity is removed. These need to be enforced via migrations in PostgreSQL,
+  but not in Cassandra.
+* Default values can be set with the `default` attribute. When a field with a default value is not set on
+  insert, the value is set via Lua, never by the underlying database. It is thus not recommended to set
+  any default values on the databases themselves, for example on migrations.
+* `required = true` works as expected. The DAO will throw an error when attempting to insert an entity
+  lacking required fields.
+* `unique = true` will mark fields so that they can not have the same value in two entities (note
+  that this might require special handling via migrations)
+* `auto = true` will:
+  * replace `null` values with a random UUID when inserting fields of type `uuid`
+  * replacing `null` values with a random string when inserting fields of type `string`
+  * adding the current time when appropriate in fields named `created_at` or `updated_at`
+
+To learn more about schemas, see:
+
+* [typedefs.lua](https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua),
```

Comment:
Better link to a specific tag (vs. the master branch). Each page of the documentation has access to the major version it is documenting (1.0.x in this case).

Reply:
I'd rather not do that because it's very easy to copy/paste this to a new version (i.e. when creating the 1.1.x docs) and then forgetting to change the links.

Reply:
That's why I pointed out the version variable: `{{page.kong_version}}` see other plugin-development guide pages

Reply:
Understood. I made the change requested every time `master` was referenced (had to use `{{page.kong_version | replace: "x", "0"}}` instead, to transform `1.0.x` into `1.0.0`).